### PR TITLE
Tried to clean up the character class by grouping functions of similar type

### PR DESCRIPTION
**Pull request by mattpott**
A small tidy up of the character class trying to group functions in a logical way.

### CODE REVIEW

**Comment by Grenkin1988** [comment]
1. No single letter variables
2. We should not use `_` in names

**Comment by mattpott** [comment]
@Grenkin1988 Can you tell me if there are anymore and if so what line numbers?

**Comment by bjubes** [comment]
`Update_Needs()` line 743
`Update_DoMovement()`752

I think they were orignally underscored to match the Lua syntax of how updates are called

**Comment by mattpott** [comment]
I will also go through and add comments inline with the recent discussions on documenting public functions.

File: Assets/Scripts/Models/Character.cs [hunk 1]
```diff
@@ -716,14 +738,140 @@ private void Update_DoJob(float deltaTime)
                 MyJob.DoWork(deltaTime);
             }
         }
+    }
 
+    private void Update_Needs(float deltaTime)
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Character.cs]: `UpdateNeeds`

File: Assets/Scripts/Models/Character.cs [hunk 2]
```diff
@@ -716,14 +738,140 @@ private void Update_DoJob(float deltaTime)
...
         // Calculate needs.
-        foreach (Need n in needs)
+        foreach (Need need in needs)
         {
-            n.Update(deltaTime);
+            need.Update(deltaTime);
         }
     }
+        
+    private void Update_DoMovement(float deltaTime)
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Character.cs]: `DoMovement`

File: Assets/Scripts/Models/Character.cs [hunk 3]
```diff
@@ -344,6 +344,8 @@ public void Update(float deltaTime)
     {
         Update_DoJob(deltaTime);
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Character.cs]: `DoJob`

**Comment by mattpott** [comment]
@Grenkin1988 Will fix these when I get home.

File: Assets/Scripts/Models/Character.cs [hunk 4]
```diff
@@ -683,6 +691,20 @@ private void GetNewJob()
...
     }
 
+    private void OnJobStopped(Job job)
+    {
+        // Job completed (if non-repeating) or was cancelled.
+        job.OnJobStopped -= OnJobStopped;
+
+        if (job != MyJob)
+        {
+            Debug.ULogErrorChannel("Character", "Character being told about job that isn't his. You forgot to unregister something.");
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Character.cs]: about -> abort

**Comment by Grenkin1988** [comment]
Merged. Would be really nice if somebody would fix naming in `Character` later

**Pull request merged**